### PR TITLE
Explicit dns servers

### PR DESCRIPTION
**Pull request by dholms**
Add ability to set up explicit dns servers to query for handle dns resolution

### CODE REVIEW

File: packages/identity/src/handle/index.ts [hunk 1]
```diff
@@ -57,4 +59,39 @@ export class HandleResolver {
       return undefined
     }
   }
+
+  async resolveDnsBackup(handle: string): Promise<string | undefined> {
```
Comment (devinivy) [inline packages/identity/src/handle/index.ts:63]: I think that if any errors happen here that we want to `return undefined`, similar to the other resolver methods.

File: packages/identity/src/handle/index.ts [hunk 2]
```diff
@@ -57,4 +59,39 @@ export class HandleResolver {
...
+    return found[0].slice(PREFIX.length)
+  }
+
+  private async getBackupNameserverIps(): Promise<string[] | undefined> {
+    if (!this.backupNameservers) {
+      return undefined
+    } else if (!this.backupNameserverIps) {
+      const res = await Promise.all(
+        this.backupNameservers.map((h) => dns.lookup(h)),
+      )
```
Comment (devinivy) [inline packages/identity/src/handle/index.ts:92]: Could also do an `allSettled()` here, take the passing ones, and log the failing ones.  If you have a bad one I think that it just causes the dns lookups to occur again and `resolveDnsBackup()` to throw each time it's called.

**Review by devinivy — COMMENTED** [review]
Just a small note on error handling, beyond that lookin good 👍

File: packages/identity/src/handle/index.ts (hunk 1 above)
Comment (dholms) [reply packages/identity/src/handle/index.ts:63]: ah yeah probs smart

**Pull request merged**